# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/media/webrtc/libwebrtcglue/WebrtcVideoCodecFactory.cpp
# Commit: c34ea4d71235
# Full Hash: c34ea4d71235c0519020893ba04d7206e7fe4f14
# Author: Michael Froman <mfroman@mozilla.com>
# Date: 2021-11-01 18:17:33
# Regressor Bug: 1654112
# File Overlap Count: 1
# Description:
#   Bug 1654112 - fix tsan error, using a destroyed mutex. r=ng
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D121007
# ==============================================================================

diff -r f547e3673690 -r c34ea4d71235 dom/media/webrtc/libwebrtcglue/WebrtcVideoCodecFactory.cpp
--- a/dom/media/webrtc/libwebrtcglue/WebrtcVideoCodecFactory.cpp	Wed Jul 21 07:51:58 2021 -0700
+++ b/dom/media/webrtc/libwebrtcglue/WebrtcVideoCodecFactory.cpp	Mon Jul 26 16:42:40 2021 -0500
@@ -23,8 +23,8 @@
     case webrtc::VideoCodecType::kVideoCodecH264: {
       // Get an external decoder
       auto gmpDecoder = WrapUnique(GmpVideoCodec::CreateDecoder(mPCHandle));
-      mCreatedGmpPluginEvent.Forward(*gmpDecoder->InitPluginEvent());
-      mReleasedGmpPluginEvent.Forward(*gmpDecoder->ReleasePluginEvent());
+      mCreatedGmpPluginEvent->Forward(*gmpDecoder->InitPluginEvent());
+      mReleasedGmpPluginEvent->Forward(*gmpDecoder->ReleasePluginEvent());
       decoder.reset(gmpDecoder.release());
       break;
     }
@@ -117,8 +117,8 @@
     case webrtc::VideoCodecType::kVideoCodecH264: {
       // get an external encoder
       auto gmpEncoder = WrapUnique(GmpVideoCodec::CreateEncoder(mPCHandle));
-      mCreatedGmpPluginEvent.Forward(*gmpEncoder->InitPluginEvent());
-      mReleasedGmpPluginEvent.Forward(*gmpEncoder->ReleasePluginEvent());
+      mCreatedGmpPluginEvent->Forward(*gmpEncoder->InitPluginEvent());
+      mReleasedGmpPluginEvent->Forward(*gmpEncoder->ReleasePluginEvent());
       encoder.reset(gmpEncoder.release());
       break;
     }